src/NextGen/DonationForm/DataTransferObjects/DonationFormQueryData.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Give\NextGen\DonationForm\DataTransferObjects;
+
+use DateTimeInterface;
+use Give\Framework\Support\Facades\DateTime\Temporal;
+use Give\NextGen\DonationForm\Models\DonationForm;
+use Give\NextGen\DonationForm\ValueObjects\DonationFormMetaKeys;
+use Give\NextGen\DonationForm\ValueObjects\DonationFormStatus;
+use Give\NextGen\Framework\Blocks\BlockCollection;
+
+class DonationFormQueryData
+{
+    /**
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var string
+     */
+    public $title;
+
+    /**
+     * @var array
+     */
+    public $settings;
+
+    /**
+     * @var DateTimeInterface
+     */
+    public $createdAt;
+
+    /**
+     * @var DateTimeInterface
+     */
+    public $updatedAt;
+
+    /**
+     * @var DonationFormStatus
+     */
+    public $status;
+
+    /**
+     * @var BlockCollection
+     */
+    public $blocks;
+
+    /**
+     * Convert data from object to Donation Form
+     *
+     * @unreleased
+     *
+     * @param  object  $queryObject
+     *
+     * @return DonationFormQueryData
+     */
+    public static function fromObject($queryObject): DonationFormQueryData
+    {
+        $self = new static();
```

Comment:
History lesson...

The `QueryData` model DTO's were created back when there was no query builder in core and models were just a twinkle in our eye.  Their main purpose was to clean up all the duplicated places we were converting data from the database into what would eventually be model instances.  At that time they served multiple purposes and were used more traditionally as DTOs.

Fast-forward to the point in which models became stable and we created a `ModelQueryBuilder` which further cleaned up all the data transferring from the different repository calls into a one-time method in the models called `fromQueryBuilderObject`.  When I created that, I made sure the method doesn't care _how_ you convert to data, it just wants to return a model instance.  So for the sake of time, those DTO's (that were already battle-tested) were simply reused inside that model method. At this point they're really just glorified actions that take in an object and return a model - but i'm really happy that they became a one-to-one in terms of object properties of models because it's going to be so easy to move all that logic somewhere else.

Reply:
That's cool! Thanks for explaining that! Sounds like it was a solid way of transitioning the existing code over to the models. 👍 

At this point, it looks like the DTOs are only used by the models and for the models. As such my question remains since the DTO ends up being a duplicate of the Model itself — which feels unnecessary. I don't see it as a tradition worth carrying on into a completely new Model. Any reason we can't move this into the factory as I was thinking, or something else similar?

Reply:
Yeah, I can move it into an action.  At least for now, I want to make sure the data transferring is as isolated as possible and avoid adding what would look like a common framework method to the factory in this add-on.

Reply:
I'm cool with an action. I got thinking about the factory we have and wondered if it could be an interesting utility beyond just making mock versions of models. But that should be another discussion for another PR.

Reply:
Cool, all set.  e24bda5c